src/renderer/base/renderer.cpp

```diff
@@ -1106,10 +1109,9 @@ bool Renderer::_isInHoveredInterval(const til::point coordTarget) const noexcept
 // - <none>
 void Renderer::_PaintCursor(_In_ IRenderEngine* const pEngine)
 {
-    const auto cursorInfo = _GetCursorInfo();
-    if (cursorInfo.has_value())
+    if (_currentCursorOptions.has_value())
     {
-        LOG_IF_FAILED(pEngine->PaintCursor(cursorInfo.value()));
+        LOG_IF_FAILED(pEngine->PaintCursor(_currentCursorOptions.value()));

```

Comment:
nit: can use `*_currentCursorOptions` (i think it skips a check where `.value()` makes a check? but we just checked)

Reply:
The compiler will be able to optimize redundant trivially inlinable branches away _if_ there isn't a call in between that it can't inspect (the compiler must assume that memory may have mutated at any time during an external call). In this case it'll be able to optimize away the check for sure (just checked it), however this still isn't the case for Debug builds and so tl;dr: Yeah 100% agreed.